pkg/registry/resource/resourceclaim/strategy.go

```diff
@@ -345,3 +347,42 @@ func dropDeallocatedStatusDevices(newClaim, oldClaim *resource.ResourceClaim) {
 }
 
 // TODO: add tests after partitionable devices is merged (code conflict!)
+
+// dropDisabledDRADeviceBindingConditionsFields removes fields which are covered by a feature gate.
+func dropDisabledDRADeviceBindingConditionsFields(newClaim, oldClaim *resource.ResourceClaim) {
+	if utilfeature.DefaultFeatureGate.Enabled(features.DRADeviceBindingConditions) && utilfeature.DefaultFeatureGate.Enabled(features.DRAResourceClaimDeviceStatus) ||
+		draBindingConditionsFeatureInUse(oldClaim) {
+		return
+	}
+
+	klog.Info("drop disabled DRADeviceBindingConditions fields")
+	if newClaim.Status.Allocation == nil {
+		return
+	} else {
+		newClaim.Status.Allocation.AllocationTimestamp = nil
+	}
```

Comment:
```suggestion
	
	newClaim.Status.Allocation.AllocationTimestamp = nil
```

nit: with `allocation := newClaim.Status.Allocation` before the if check the code becomes a bit shorter.